internal/remotestorage/remotestorage.go

```diff
@@ -0,0 +1,65 @@
+package remotestorage
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+// RemoteStorage defines functions to interact with application storage service
+type RemoteStorage interface {
+	Upload(url, fileName, formType string) ([]byte, error)
+}
+
+// Storage implements functions for RemoteStorage interface
+type Storage struct {
+	HTTPClient *http.Client
+}
+
+// NewRemoteStorage returns an implementation for RemoteStorage
+func NewRemoteStorage() RemoteStorage {
+	return &Storage{
+		HTTPClient: &http.Client{},
+	}
+}
+
+// Upload uploads file to remote storage
+func (s *Storage) Upload(url, fileName, formType string) ([]byte, error) {
```

Comment:
Might be worth splitting this method a bit. You already commented the different steps/parts, so those would be good candidates!

Reply:
Sure. I'll split this method.